src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs

```diff
@@ -1453,15 +1451,15 @@ protected virtual void OnMouseDragMove(int x, int y)
     protected virtual void OnMouseEnter()
     {
         Control ctl = Control;
-        Control parent = ctl;
-        object parentDesigner = null;
+        Control? parent = ctl;
+        object? parentDesigner = null;
 
         while (parentDesigner is null && parent is not null)
         {
             parent = parent.Parent;
             if (parent is not null)
             {
-                object d = _host.GetDesigner(parent);
+                object? d = _host?.GetDesigner(parent);
```

Comment:
Let's go ahead and spell out the variable name
```suggestion
                object? designer = _host?.GetDesigner(parent);
```